application/controllers/template_controller.py

```diff
@@ -0,0 +1,64 @@
+from application.models.models import CommunicationTemplate
+from application.utils.session_wrapper import with_db_session
+from application.utils.exceptions import InvalidTemplateException, DatabaseError
+from application.models.schema import template_schema
+from jsonschema import validate, ValidationError
+from structlog import get_logger
+
+
+logger = get_logger()
+
+UPLOAD_SUCCESSFUL = 'The upload was successful'
+INVALID_TEMPLATE = 'Template is invalid'
+PREEXISTING_TEMPLATE = 'Id already exists'
+
+
+def get_template_by_id(template_id, session):
+    return session.query(CommunicationTemplate).filter(CommunicationTemplate.id == template_id).first()
```

Comment:
This needs more defensive coding. As a minimum, catching the base SqlAlchemyError (http://docs.sqlalchemy.org/en/latest/core/exceptions.html#sqlalchemy.exc.SQLAlchemyError) exception class should be carried out here, and around any other SQLAlchemy session methods.

This should be in addition to the `@with_db_session` decorator defined in the `session_wrapper.py` module.

Reply:
We should relook at the with_db_session approach as discussed earlier around just creating transactions around the requests